terraform/southpark.tf

```diff
@@ -0,0 +1,170 @@
+provider "aws" {
+  version = "~> 2.0"
+  region  = "us-east-2"
+}
+
+
+#Create repo
+resource "aws_ecr_repository" "southpark_repo" {
+  name = "southpark_repo" # Naming my repository
+}
+
+# Providing a reference to our default VPC
+resource "aws_default_vpc" "default_vpc" {
+}
+
+# Providing a reference to our default subnets
+resource "aws_default_subnet" "default_subnet_a" {
+  availability_zone = "us-east-2a"
+}
+
+resource "aws_default_subnet" "default_subnet_b" {
+  availability_zone = "us-east-2b"
+}
+
+resource "aws_default_subnet" "default_subnet_c" {
+  availability_zone = "us-east-2c"
+}
+
+resource "aws_ecs_cluster" "southpark_cluster" {
+  name = "southpark_cluster" # Naming the cluster
+}
+
+#Create role
+resource "aws_iam_role_policy_attachment" "ecsTaskExecutionRole_policy" {
+  role       = "${aws_iam_role.ecsExecutionRole.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "ecsExecutionRole" {
+  name               = "ecsExecutionRole"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role_policy.json}"
+}
+
+
+#Create task
+resource "aws_ecs_task_definition" "southpark_task" {
+  family                   = "southpark_task" # Naming our first task
+  container_definitions    = <<DEFINITION
+  [
+    {
+      "name": "southpark_task",
+      "image": "${aws_ecr_repository.southpark_repo.repository_url}",
+      "essential": true,
+      "portMappings": [
+        {
+          "containerPort": 3000,
+          "hostPort": 3000
+        }
+      ],
+      "memory": 512,
+      "cpu": 256
+    }
+  ]
+  DEFINITION
+  requires_compatibilities = ["FARGATE"] # Stating that we are using ECS Fargate
```

Comment:
I used Fargate just so it would manage the cluster for me. I tried doing it without any cluster but had no luck with that

Reply:
OK, I just did the computation on c5 plain ec2 instances and there FARGATE is not "quite more expensive" but rather 20% more expensive. Thus, it might be fine to use FARGATE.